### DISCUSSION TITLE
v0.2.3

### DISCUSSION THREAD

**Discussion by ziatdinovmax** [ziatdinovmax/pyroVED#43] [Q&A]
### New functionalities

1) The ability to condition (i)VAE on both continuous and discrete variables. The latter usually represent categorical data classes (aka labels) whereas the former can represent some continuous property (or multiple properties) of the data that we *know*. See the example [notebook](https://colab.research.google.com/github/ziatdinovmax/pyroVED/blob/main/examples/pyroVED_examples_C_cont_VAE.ipynb).

2) (i)VAE for semi-supervised regression. To date, most applications of the semi-supervised VAE were for categorical data. Here we introduce an option to use semi-supervised VAE for regression analysis where for some (small) part of the data both the label variable and the predictors are observed, while for other (larger) part of the data only the predictors are given. The usage of the ```ss_reg_iVAE``` [class](https://github.com/ziatdinovmax/pyroVED/blob/main/pyroved/models/ss_reg_ivae.py) is similar to that of the ss-iVAE.

3) The ```auxSVITrainer``` now works for both classification and regression tasks. To choose between the two, use the ```task``` argument (e.g. ```task="classification" or ```task=regression```).

### Breaking changes
1) The way categorical variables are passed to ```ivae.manifold2d``` has changed.

Before:
```python
for i in range(10):
    cvae.manifold2d(d=12, label=i, cmap="viridis")
```
Now
```python
for cls in range(10):
    cls = pv.utils.to_onehot(torch.tensor([cls,]), 10)
    cvae.manifold2d(d=12, y=cls, cmap="viridis")
```
2) pyroVED < 0.2.3 is not compatible with the latest PyTorch version 1.9. Hence, if using earlier pyroVED versions, downgrade the PyTorch < 1.9. See [this page](https://pytorch.org/get-started/previous-versions/) for details.

<hr /><em>This discussion was created from the release <a href='https://github.com/ziatdinovmax/pyroVED/releases/tag/v0.2.3'>v0.2.3</a>.</em>
